graphql.go

```diff
@@ -131,16 +131,16 @@ func (c *Client) runWithJSON(ctx context.Context, req *Request, resp interface{}
 	if err != nil {
 		return err
 	}
+	if res.StatusCode != http.StatusOK {
```

Comment:
This should come after `defer res.Body.Close()`. As mentioned in [the http.Client.Do docs](https://golang.org/pkg/net/http/#Client.Do):

> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close.

Similar below.